From d0ce50ec55ba0eb8450507aa798fd0ffb3ddecde Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Sat, 29 Apr 2017 17:55:17 +0800
Subject: [PATCH 60/69] arm64: dts: allwinner: Add Ethernet support for
 BananaPi M64

Add the required DT magic to enable Ethernet on the board.
This includes the actual Ethernet MAC node, referring the RGMII
pins of the device.

This is lifted from the respective Orange Pi Plus patch by
Corentin Labbe <clabbe.montjoie@gmail.com>.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 .../arm64/boot/dts/allwinner/sun50i-a64-bananapi-m64.dts | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-a64-bananapi-m64.dts b/arch/arm64/boot/dts/allwinner/sun50i-a64-bananapi-m64.dts
index c9ee8afd1bd1..d13bdb238ee7 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-a64-bananapi-m64.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-a64-bananapi-m64.dts
@@ -53,6 +53,7 @@
 	aliases {
 		serial0 = &uart0;
 		serial1 = &uart1;
+		ethernet0 = &emac;
 	};
 
 	chosen {
@@ -76,6 +77,15 @@
 	status = "okay";
 };
 
+&emac {
+	pinctrl-names = "default";
+	pinctrl-0 = <&emac_rgmii_pins>;
+	phy-handle = <&ext_phy>;
+	phy-supply = <&reg_dc1sw>;
+	phy-mode = "rgmii";
+	status = "okay";
+};
+
 &i2c1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&i2c1_pins>;
@@ -90,6 +100,12 @@
 	status = "okay";
 };
 
+&mdio {
+	ext_phy: ethernet-phy@1 {
+		reg = <1>;
+	};
+};
+
 &mmc0 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&mmc0_pins>;
-- 
2.12.2

